### PR TITLE
Feature/midi player

### PR DESCRIPTION
**Pull request by femans**
implements midi player with random instruments.
Great care has been taken to make sure the player stops properly, does not play 2 songs through eachother, gets torn down when moving to a next song, and cancels the download if you stop/skip before it was (fully) loaded.

also fixes 2 bugs in the playlist functionality that were found along the way and makes the avatar icons on playlist prettier.

### CODE REVIEW

**Review by dokterbob — CHANGES_REQUESTED** [review]
Some small comments, do with it as you will.

MIDI and FLAC and MP3 all play wonderful, also going smoothly from song to song, the UX is also responsive and sensible. Consider the issue below minor issues, deserving of a ticket but not requiring fixup in this PR. 

The code comments might be worth a small fixup though (e.g. removing commented code or TODO's).

## Formats tested (Brave)
- FLAC
- MP3
- MIDI
- WMA
- WAV
- OPUS

## User test: issues
- [x] Prev/next not working from detail view (entering from list view)
- [x] No user feedback for unsupported files
- [x] mdi-chevron-up missing
- [ ] No file type shown in list and detail header for OPUS

### Prev/next not working from detail view 
1. Open audio list view https://bafybeidrmheunuh7awdmfvr23o7yll4exi65uzlstgb7vskz2fz6giw5ji.on.fleek.co/#/search?type=audio&page=1
2. Click on a song, moving to detail view
3. Click play
4. Click next or prev
 
Expected: move to previous or next song in search results.
Instead: nothing happens.

Perhaps this is intended behaviour, it just doesn't match my naive expectations.

### No user feedback for unsupported files
On detail view for an unspported file, clicking on play results in an error in the console but no user feedback.

Example: https://bafybeidrmheunuh7awdmfvr23o7yll4exi65uzlstgb7vskz2fz6giw5ji.on.fleek.co/#/search/detail/audio/QmcLYjMMrAnou6LAgamY38wiqFgJhdyJ4E5RAEgQTxptGr?type=audio&page=2&last_seen=%3C30d

### mdi-chevron-up missing
In detail view, I am getting the following error in the console:
`Error: <path> attribute d: Expected number, "mdi-chevron-up".`
Example: https://bafybeidrmheunuh7awdmfvr23o7yll4exi65uzlstgb7vskz2fz6giw5ji.on.fleek.co/#/search?type=audio&page=2&last_seen=%3C30d

### No file type shown in list and detail header for OPUS, WAV
#### Example (list)
image

https://bafybeidrmheunuh7awdmfvr23o7yll4exi65uzlstgb7vskz2fz6giw5ji.on.fleek.co/#/search?type=audio&page=3&last_seen=%3C30d

#### Example (detail)
https://bafybeidrmheunuh7awdmfvr23o7yll4exi65uzlstgb7vskz2fz6giw5ji.on.fleek.co/#/search/detail/audio/QmT5mBGTfgh863gg28bs7pED2sXXKukXuE8vaYx3UyD99L?type=audio&page=3&last_seen=%3C30d

File: src/helpers/midiPlayer.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  private patchBay: Record<number, number>;
+  private trackNames: Record<number, string>;
+  private midiPlayer;
+  private loaded = false;
+  private handleMidiEvent = (event: IMidiEvent) => {
+    if (event.name === "Sequence/Track Name") {
+      this.trackNames[event.track] = event.string?.toLowerCase();
+    } else if (event.name === "Note on" && event.velocity > 0) {
+      if (!this.patchBay[event.track]) {
+        // eslint-disable-next-line no-nested-ternary
```
[the remark this answers is not in the document]
Comment (femans) [reply src/helpers/midiPlayer.ts:49]: good point; I literally copy/pasted this from my hackjob of half a year ago and it worked so i left it alone

File: src/interfaces/audioInterfaces.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { IFile } from "./IFile";
+
+export interface IPlaylist {
+  index?: number | undefined;
+  entries: IFile[];
+}
+
+export type MediaPlayerEvent = "playerror" | "loaderror" | "end" | "load" | "play" | "pause";
+
+// TODO: expand this to a more generic player interface (now it is a shadow of the Howl player)
```
[the remark this answers is not in the document]
Comment (femans) [reply src/interfaces/audioInterfaces.ts:10]: https://github.com/ipfs-search/dweb-search-frontend/issues/255

**Comment by femans** [comment]
@dokterbob created issues for everything mentioned and cleaned up the instrument selector code, should be good now

File: src/helpers/midiPlayer.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (femans) [reply src/helpers/midiPlayer.ts]: Now it is nicely organized code

**Review by dokterbob — APPROVED** [review]

**Pull request merged**